tusclient/request.py

```diff
@@ -69,6 +70,7 @@ def perform(self):
         Perform actual request.
         """
         self.handle.perform()
+        self._status_code = self.handle.getinfo(pycurl.RESPONSE_CODE)
```

Comment:
if we make `self.status_code` the attribute as suggested, then we would change this line to 
`self._status_code = self.handle.getinfo(pycurl.RESPONSE_CODE)`